src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java

```diff
@@ -756,21 +765,26 @@ private static int compareImports(String import1, String import2) {
     }
 
     /**
-     * Counts empty lines before given.
-     * @param lineNo
-     *        Line number of current import.
-     * @return count of empty lines before given.
+     * Counts empty lines between given parameters.
+     * @param fromLineNo
+     *        One-based line number of previous import.
+     * @param toLineNo
+     *        One-based line number of current import.
+     * @return count of empty lines between given parameters.
      */
-    private int getCountOfEmptyLinesBefore(int lineNo) {
+    private int getCountOfEmptyLinesBetween(int fromLineNo, int toLineNo) {
         int result = 0;
         final String[] lines = getLines();
-        //  [lineNo - 2] is the number of the previous line
-        //  because the numbering starts from zero.
-        int lineBeforeIndex = lineNo - 2;
-        while (lineBeforeIndex >= 0
-                && CommonUtil.isBlank(lines[lineBeforeIndex])) {
-            lineBeforeIndex--;
-            result++;
+
+        //  "- 2" because the numbering starts from zero
+        for (int i = toLineNo - 2; i > fromLineNo - 2; i--) {
+            if (CommonUtil.isBlank(lines[i])) {
+                result++;
+            }
+            else {
+                // stop on non-empty line to keep the behavior same as previous
```

Comment:
code is always inn current state .... there is no previous behavior.

if that is not good behavior .. please define `till issue https://.......` 
if that is expected, please remove "to keep the behavior same as previous"

Reply:
There’s an ralated issue https://github.com/checkstyle/checkstyle/issues/7314 but has not been approved yet. Can you please take a look at it before I fix the comment?

Reply:
Fixed :)

Reply:
good for now.